stdlib/Compflags

```diff
@@ -34,5 +34,6 @@ case $1 in
   stdlib__scanf.cmx) echo ' -inline 9';;
   *Labels.cm[ox]) echo ' -nolabels -no-alias-deps';;
   stdlib__float.cm[ox]) echo ' -nolabels -no-alias-deps';;
+  stdlib__oo.cmi) echo ' -no-principal';;
```

Comment:
I'm slightly surprised by this.

Reply:
If you remove it, the type of `Oo.copy` loses abbreviations (i.e. applying it to an object returns a non-abbreviated type).
I admit that this tuning is disturbing, but losing the abbreviation because of a change in the standard library seems a nuisance.

Reply:
It would be nice to leave a comment here so that we don't remove (or consider removing) the flag in the future because we forgot the reason for having it.

(Is there a different type signature that one could use to enforce sharing even in `-principal` mode?)

Reply:
Unfortunately, I see no way to enforce the sharing of an alias inside a signature in principal mode. The unsharing of aliases is part of the design.
There may be a way by using a different way to introduce sharing, but it is going to be subtle. Even something like
```ocaml
type 'a identity = 'a -> 'a
val copy : < .. > identity
```
mysteriously doesn't seem to work.